File: Introibo/Data/ReferenceEntry.swift
import Foundation

// Matches Introibo/Resources/reference.json. Ported from
// prototype/reference.html's REFERENCE object. All fields except
// slug/title/cat/summary are optional per-entry.

struct ReferenceEntry: Identifiable, Decodable, Hashable {
    let slug: String
    var title: String
    let latin: String?
    let cat: String                 // Category label
    var summary: String
    var history: String?
    var practice: String?
    var notes: String?
    var scripture: Scripture?
    var related: [RelatedLink]? = nil

    var id: String { slug }

    struct Scripture: Decodable, Hashable {
        let ref: String
        let lat: String
        var eng: String
    }
}

File: android/app/src/main/java/com/lampstandhq/introibo/data/model/ReferenceEntry.kt
package com.lampstandhq.introibo.data.model

import com.lampstandhq.introibo.data.links.RelatedLink
import kotlinx.serialization.Serializable

/**
 * Matches reference.json. All fields except slug/title/cat/summary
 * are optional per-entry.
 */
@Serializable
data class ReferenceEntry(
    val slug: String,
    val title: String,
    val latin: String? = null,
    val cat: String,                 // Category label
    val summary: String,
    val history: String? = null,
    val practice: String? = null,
    val notes: String? = null,
    val scripture: Scripture? = null,
    val related: List<RelatedLink>? = null,
) {
    @Serializable
    data class Scripture(
        val ref: String,
        val lat: String,
        val eng: String,
    )
}
